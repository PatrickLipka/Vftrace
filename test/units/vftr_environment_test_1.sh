#!/bin/bash
set -x
ref_out_dir=${srcdir}/ref_output
ref_in_dir=${srcdir}/ref_input
testname=vftr_environment_test_1
outfile=$testname.out

if [ -f $outfile ] ; then
   rm -f $outfile
fi

# The user might have set some VFTR_ environment variables.
# We save them in an array and unset them all.
# After the test, we reset them to their original value.
vftr_variables=(`env | grep VFTR_`)
for v in ${vftr_variables[@]}; do
  unset `echo $v | cut -f1 -d "="`
done
./test_vftrace $testname
diff $ref_out_dir/$outfile $outfile

# Reset all environment variables here
for v in ${vftr_variables[@]}; do
  export $v;
done
