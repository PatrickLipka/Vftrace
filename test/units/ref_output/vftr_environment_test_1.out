VFTR_OFF: true
VFTR_SAMPLING(default): false
VFTR_REGIONS_PRECISE(default): true
VFTR_OUT_DIRECTORY(default): .
VFTR_LOGFILE_BASENAME(default): (null)
VFTR_LOGFILE_ALL_RANKS(default): false
VFTR_SAMPLETIME(default): 0.01
VFTR_STOPTIME(default): 604800
VFTR_ACCURATE_PROFILE(default): false
VFTR_PROF_TRUNCATE(default): true
VFTR_MPI_LOG(default): false
VFTR_MPI_SHOW_SYNC_TIME(default): false
VFTR_SIGNALS_OFF(default): true
VFTR_BUFSIZE(default): 8
VFTR_ACCURATE_PROFILE(default): false
VFTR_DETAIL_UNTIL_CUM_CYCLES(default): 90.00
VFTR_SCENARIO_FILE(default): (null)
VFTR_LICENSE_VERBOSE(default): false
VFTR_PRINT_STACKS_FOR(default): (null)
VFTR_PRINT_LOADINFO_FOR(default): (null)
***************************
VFTR_OFF: false
VFTR_SAMPLING: true
VFTR_REGIONS_PRECISE: false
VFTR_OUT_DIRECTORY: "foo/bar"
VFTR_LOGFILE_BASENAME(default): (null)
VFTR_LOGFILE_ALL_RANKS(default): false
VFTR_SAMPLETIME: 12.34
VFTR_STOPTIME(default): 604800
VFTR_ACCURATE_PROFILE(default): false
VFTR_PROF_TRUNCATE(default): true
VFTR_MPI_LOG: true
VFTR_MPI_SHOW_SYNC_TIME(default): false
VFTR_SIGNALS_OFF(default): true
VFTR_BUFSIZE: 1234
VFTR_ACCURATE_PROFILE(default): false
VFTR_DETAIL_UNTIL_CUM_CYCLES(default): 90.00
VFTR_SCENARIO_FILE(default): (null)
VFTR_LICENSE_VERBOSE(default): false
VFTR_PRINT_STACKS_FOR(default): (null)
VFTR_PRINT_LOADINFO_FOR(default): (null)
